icefoganalytics/internal-data-portal
# Add Basic Login Setup To App

Fixes https://github.com/icefoganalytics/internal-data-portal/issues/12

Relates to:
- [0001, Data Modeling, 2024-01-05](https://docs.google.com/document/d/1Mdbl_Cy0BgKiq0DGdvOKUqJn-FdgbF0LSVykwf__FY8/edit?pli=1)
- https://github.com/icefoganalytics/elcc-data-management/blob/main/web/src/modules/authentication/views/SignIn.vue

# Context

Add login to the app. This requires
- a front-end Auth0 setup, 
- and a back-end Auth0 setup
- basic user modeling in the database (see data modeling document)
- users should be auto-create when pulled from auth0 in the back-end

# Implementation

Add secure by default back-end api routes.
Add user auto creation via back-end middleware, this seems like its not a great pattern, but I don't want to completely change the pattern from the other apps today.
Add secure by default front-end route guards.
Auto-load current user at application level in front-end.

# Screenshots

Dashboard demonstrating successful login
![image](https://github.com/icefoganalytics/internal-data-portal/assets/23045206/0bf73508-e626-4fff-a21a-e33cc71207c4)

# Testing Instructions

1. Boot the app via `dev up`
2. Go to the app at http://localhost:8080, you will be redirected to the Auth0 sign in.
3. After sign in check that you are redirected to the /dashboard.
4. Check that the back end serves up the current user in the /dashboard.
5. Go to the http://localhost:8080/status page and check that the "return to" links to the dashboard.
6. Go back to the dashboard via that link.
7. Log out from the bottom of the /dashboard.
8. Go back to the http://localhost:8080/status page and check that the "return to" links to the "sign in" page.
